src/Symfony/Component/Serializer/Mapping/AttributeMetadata.php

```diff
@@ -200,8 +246,16 @@ public function merge(AttributeMetadataInterface $attributeMetadata): void
 
         // Overwrite only if not defined
         $this->maxDepth ??= $attributeMetadata->getMaxDepth();
-        $this->serializedName ??= $attributeMetadata->getSerializedName();
-        $this->serializedPath ??= $attributeMetadata->getSerializedPath();
+
+        // Overwrite only if both serialized names and paths are empty
+        if (!$this->serializedNames && !$this->serializedPaths) {
+            $this->serializedNames = method_exists($attributeMetadata, 'getSerializedNames')
+                ? $attributeMetadata->getSerializedNames()
+                : array_filter(['*' => $attributeMetadata->getSerializedName()]);
```

Comment:
I don't think `array_filter` is the solution here. Indeed, `var_dump(array_filter(['*' => '0']));` will return `[]`, which can lead to bugs if anybody has a `0` group.
(same for `DebugCommand`)